massa-pos-exports/src/pos_final_state.rs

```diff
@@ -359,4 +386,114 @@ impl PoSFinalState {
         }
         Some(index)
     }
+
+    /// Gets a cycle of the Proof of Stake `cycle_history`. Used only in the bootstrap process.
+    ///
+    /// # Arguments:
+    /// `cursor`: indicates the bootstrap state after the previous payload
+    ///
+    /// # Returns
+    /// The PoS cycle and the updated cursor
+    pub fn get_cycle_history_part(
+        &self,
+        cursor: StreamingStep<u64>,
+    ) -> Result<(Option<CycleInfo>, StreamingStep<u64>), ModelsError> {
+        let cycle_index = match cursor {
+            // FOLLOW-UP TODO: use the config value for `6`
+            StreamingStep::Started => usize::from(self.cycle_history.len() >= 6),
+            StreamingStep::Ongoing(last_cycle) => {
+                if let Some(index) = self.get_cycle_index(last_cycle) {
+                    if index == self.cycle_history.len() - 1 {
+                        return Ok((None, StreamingStep::Finished));
+                    }
+                    index.saturating_add(1)
+                } else {
+                    return Err(ModelsError::OutdatedBootstrapCursor);
+                }
+            }
+            StreamingStep::Finished => return Ok((None, cursor)),
+        };
+        let cycle_info = self
+            .cycle_history
+            .get(cycle_index)
+            .expect("a cycle should be available here");
+        Ok((
+            Some(cycle_info.clone()),
+            StreamingStep::Ongoing(cycle_info.cycle),
+        ))
+    }
+
+    /// Gets a part of the Proof of Stake `deferred_credits`. Used only in the bootstrap process.
+    ///
+    /// # Arguments:
+    /// `cursor`: indicates the bootstrap state after the previous payload
+    ///
+    /// # Returns
+    /// The PoS `deferred_credits` part and the updated cursor
+    pub fn get_deferred_credits_part(
```

Comment:
The streaming step never go to finished in this function

Reply:
It does